### DISCUSSION TITLE
Laravel Nova auth login with custom email,password db column

### DISCUSSION THREAD

**Discussion by Ruitjes** [laravel/nova-issues#4516] [Q&A] [answered]
Hello, 

I'm trying to add Nova to my already existing non-laravel PHP application. But for the users table i'm using different db columns than the default ones. My question how can I change these default values for e.g: email -> u_emailadres or password -> u_sleutel

I've already tried to add an accessor to my Users model but with no luck.

`        public function email(): Attribute {
        return Attribute::get(fn() => $this->u_emailadres);
    }
`

Error logging:

`[2022-06-30 20:54:52] local.ERROR: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'email' in 'where clause' (SQL: select * from users where email = admin@admin.nl limit 1) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'email' in 'where clause' (SQL: select * from users where email = admin@admin.nl limit 1) at 
/Illuminate/Database/Connection.php:759)
[stacktrace]`

How can I change these hardcoded columns?

**Comment by anditsung** [comment]
maybe using this
https://github.com/jarektkaczyk/eloquence

**Comment by crynobone** [comment, marked as answer]
You can follow the following code: https://github.com/nova-issues/issue-4516/commit/2498ae7693587fb516ad6447d7fcab39903591eb

**Reply by Ruitjes** [reply]
@crynobone Am I missing something, I have followed these steps:

Added mutator for my email and password just like yours.


Created a custom nova-tool like:
`php artisan nova:tool vr/custom-login`

Added login controller like to Http/Controllers/LoginController:

```
<?php

namespace vr\CustomLogin\Http\Controllers;

use Illuminate\Http\Request;
use Laravel\Nova\Http\Controllers\LoginController as NovaLoginController;

class LoginController extends NovaLoginController
{
    /**
     * Get the needed authorization credentials from the request.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    protected function credentials(Request $request)
    {
        return with($request->only($this->username(), 'password'), function ($input) {
            $input['u_emailadres'] = $input[$this->username()];
            unset($input[$this->username()]);

            return $input;
        });
    }
}
```

Register too ToolServiceProvider.php
```
    /**
     * Register any application services.
     *
     * @return void
     */
    public function register()
    {
        $this->app->bind(NovaLoginController::class, Http\Controllers\LoginController::class);
    }
```

The password now listens to the `u_sleutel`, so that one works.

**Reply by Ruitjes** [reply]
What I currently have: https://github.com/Ruitjes/veiligreserveren-dashboard

**Reply by anditsung** [reply]
> What I currently have: https://github.com/Ruitjes/veiligreserveren-dashboard

check your PR.
